Practise_Problems/orderingSystem.js

```diff
@@ -1,114 +1,115 @@
+                else if (store[tracker][1] === price) {
+                  store[tracker][0].push(item);
+                  //store[tracker][0].sort();
+                  break;
+                }
+
+                //If the price is less than items within the store. Insert
+                //Split the array into 2 parts. Insert new item into the middle of the first part.
+                //Then merge them together
+                else {
+                  let firstHalf = store.slice(0, tracker);
+                  let secondHalf = store.slice(tracker);
+
+                  firstHalf.push([[item], price]);
+                  store = firstHalf.concat(secondHalf);
+                  break;
+                }
               }
 
-              //If the price is less than items within the store. Insert
-              //Split the array into 2 parts. Insert new item into the middle of the first part.
-              //Then merge them together
-              else {
-                let firstHalf = store.slice(0, tracker);
-                let secondHalf = store.slice(tracker);
-
-                firstHalf.push([[item], price]);
-                store = firstHalf.concat(secondHalf);
-                break;
-              }
+              break;
             }
-
-            break;
           }
         }
+
+        //Store is empty. First item ever
+        else {
+          store.push([item], price);
+        }
       }
 
-      //Store is empty. First item ever
+      //This is VIEW
       else {
-        store.push([item], price);
+        result.push(store[visited][0].sort());
+        visited += 1;
       }
     }
 
-    //This is VIEW
-    else {
-      result.push(store[visited][0].sort());
-      visited += 1;
-    }
-  }
-
-  console.log("-------------------COMMANDS-------------------");
-  console.log(commands);
-  console.log("-------------------STORE-------------------");
-  console.log(store);
-  console.log("-------------------RESULT VIEW-------------------");
-
-  //Showing whole result array (debug purpose)
-  return result;
-
-  //Showing the Nth cheapest item based on # of visits.
-  //return result[result.length-1];
-};
-
-let commands = [
-  ["INSERT", "Pizza", 8],
-  ["INSERT", "Milk", 4],
-  ["INSERT", "Gum", 1],
-  ["INSERT", "Coffee", 4],
-  ["VIEW", "-", "-"],
-  ["INSERT", "Snack", 5],
-  ["VIEW", "-", "-"],
-  ["VIEW", "-", "-"],
-];
-
-console.log(machine(commands));
+    console.log("-------------------COMMANDS-------------------");
+    console.log(commands);
+    console.log("-------------------STORE-------------------");
+    console.log(store);
+    console.log("-------------------RESULT VIEW-------------------");
+
+    //Showing whole result array (debug purpose)
+    return result;
+
+    //Showing the Nth cheapest item based on # of visits.
+    //return result[result.length-1];
+  };
+
+  let commands = [
+    ["INSERT", "Pizza", 8],
+    ["INSERT", "Milk", 4],
+    ["INSERT", "Gum", 1],
```

Comment:
bug: when I comment out this line (`["INSERT", "Gum", 1],`) and run the code it crashed 
```
/Users/ryansmit/Projects/DataStructureAlgorithm_Study/Practise_Problems/orderingSystem.js:86
        result.push(store[visited][0].sort());
                                      ^

TypeError: store[visited][0].sort is not a function
    at machine (/Users/ryansmit/Projects/DataStructureAlgorithm_Study/Practise_Problems/orderingSystem.js:86:39)
    at Object.<anonymous> (/Users/ryansmit/Projects/DataStructureAlgorithm_Study/Practise_Problems/orderingSystem.js:115:15)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47
```